torch/profiler/profiler.py

```diff
@@ -0,0 +1,191 @@
+import torch.autograd.profiler as prof
+from torch.autograd import ProfilerActivity
+
+from typing import Callable, Iterable, Optional
+
+class EnablePred(object):
+    """
+    EnablePred describes on which steps profiler is active:
+    - profiler starts in inactive state and stays in inactive state for the first 'wait' steps
+    - profiler then enters a warmup state and stays in this state for the next 'warmup' steps
+    - profiler then starts actively tracing/collecting stats for the next 'active' steps
+    - after this, profiler returns to the inactive state and cycle repeats
+
+    In case output_fn is specified, it is called every time the trace is ready
```

Comment:
This sounds like a callback instead of an "output_fn", is there a name involving "callback" that would fit?

Reply:
Follow-up question, what is the signature of this function?